src/clr-angular/forms/common/form.ts

```diff
@@ -28,4 +49,52 @@ export class ClrForm {
   markAsTouched() {
     this.markControlService.markAsTouched();
   }
+
+  @HostListener('submit')
+  onFormSubmit() {
+    this.markAsTouched();
+    // https://github.com/angular/angular/issues/12463
+    if (isPlatformBrowser(this.platformId)) {
+      const invalidControl = this.el.nativeElement.querySelector('.ng-invalid');
+      if (invalidControl) {
+        invalidControl.focus();
+        this.updateAriaLive();
+      } else {
+        this.resetAriaLive();
+      }
+    }
+  }
+
+  ngOnInit() {
+    // ClrForms need to have aria live element for screen readers.
+    this.ariaLiveElement = document.createElement('div');
+    this.ariaLiveElement.setAttribute('aria-live', 'polite');
+    this.ariaLiveElement.setAttribute('role', 'status');
+    this.ariaLiveElement.setAttribute('aria-atomic', 'true');
+    this.ariaLiveElement.setAttribute('aria-relevent', 'text');
+    this.ariaLiveElement.classList.add('clr-sr-only');
+    this.renderer.insertBefore(this.el.nativeElement, this.ariaLiveElement, this.el.nativeElement.firstChild);
```

Comment:
All the above DOM methods are native DOM methods, why is this one alone using Angular renderer2 here?

Reply:
I didn't realize that the renderer could work on elements that aren't in the DOM already I'll switch to use the renderer.